docs/src/dataDisplay/banner.js

```diff
@@ -421,31 +420,31 @@ export default class ViewsBanner extends React.Component {
     }
 
     _onAfterClose(id) {
-        let acData = this.state.acData;
+        const { acData } = this.state;
 
-        _.remove(this.state.acData, d => d.id === id);
+        _.remove(this.state.acData, (d) => d.id === id);
 
-        this.setState({ acData: acData });
+        this.setState({ acData });
     }
 
     _onAcBannerClick(id) {
-        let acData = this.state.acData;
-        let newAcData = _.find(acData, d => d.id === id);
+        const { acData } = this.state;
+        let newAcData = _.find(acData, (d) => d.id === id);
 
         if (_.isObject(newAcData)) {
             newAcData.isOpen = !newAcData.isOpen;
         } else {
             newAcData = {
-                id: id,
+                id,
                 isOpen: true,
                 message: 'A short notification description',
                 title: `AC Banner ${id + 1}`,
-                type: 'notification'
+                type: 'notification',
             };
             acData.push(newAcData);
         }
 
-        this.setState({ acData: acData });
+        this.setState({ acData });
```

Comment:
Probably doesn't matter terribly (it's just to power a documentation sample), but I don't think we're following a proper immutability approach around `acData` value in state.

Maybe something like this...?

```JavaScript
_onAcBannerClick(id) {
    this.setState((prevState) => {
        const { acData: prevAcData } = prevState;
        const prevAcDataItem = _.find(prevAcData, (d) => d.id === id);
        let newAcData = null;
        
        if (_.isObject(prevAcDataItem)) {
            // if `prevAcDataItem` is a single object we want a new object with `isOpen` toggled
            newAcData = {
                ...prevAcDataItem,
                isOpen: !prevAcDataItem.isOpen,
            };
        } else {
            // if `prevAcDataItem` is an array, append a new item
            const newAcDataItem {
                id: id,
                id,
                isOpen: true,
                message: 'A short notification description',
                title: `AC Banner ${id + 1}`,
                type: 'notification'
                type: 'notification',
            };
            
            newAcData = [
                ...prevAcDataItem, // existing items in the array
                newAcDataItem,
            ];
        }
        
        return { acData: newAcData };
    });
}

```

We can consider this a NAB for the purposes of an ESLINT auto-fix PR and flag it and come back to it later (if at all).

Reply:
Yeah, I would agree that this is NAB for this particular problem.
Auto fix does help with removing quite a few issues, which i think is sort of convenient, beside not returning destructured props to their own lines.

Reply:
Resolving.